compiler/src/main/java/com/squareup/anvil/compiler/codegen/reference/AnnotatedReferenceIr.kt

```diff
@@ -0,0 +1,21 @@
+package com.squareup.anvil.compiler.codegen.reference
+
+import org.jetbrains.kotlin.ir.util.isAnnotationWithEqualFqName
+import org.jetbrains.kotlin.name.FqName
+
+internal interface AnnotatedReferenceIr {
+  val annotations: List<AnnotationReferenceIr>
+
+  fun isAnnotatedWith(fqName: FqName): Boolean =
+    annotations.any { it.annotation.isAnnotationWithEqualFqName(fqName) }
```

Comment:
```suggestion
    annotations.any { it.fqName == fqName }
```
That should be enough and is faster.